astroid/nodes/node_classes.py

```diff
@@ -5514,6 +5514,7 @@ def _create_basic_elements(
     """Create a list of nodes to function as the elements of a new node."""
     elements: list[NodeNG] = []
     for element in value:
+        # NOTE: avoid accessing any attributes of element in the loop.
```

Comment:
Let's remove this comment and the one below. If the implementation of `const_factory` ever changes this is outdated. Imo, comments should be at the offending code (which you already did :smile:)